pkg/reposync/conditions.go

```diff
@@ -123,16 +123,28 @@ func setCondition(rs *v1beta1.RepoSync, condType v1beta1.RepoSyncConditionType,
 	if condition.Status != status {
 		condition.Status = status
 		condition.LastTransitionTime = lastUpdate
-		conditionTransited = true
+		condition.Reason = reason
+		condition.Message = message
+		condition.Commit = commit
+		condition.ErrorSourceRefs = errorSources
+		condition.ErrorSummary = errorSummary
+		condition.LastUpdateTime = lastUpdate
+		transitioned = true
+		updated = true
+	} else {
```

Comment:
minor nit on code style:
```golang
	if condition.Status != status {
		condition.Status = status
		condition.LastTransitionTime = lastUpdate
		transitioned = true
		updated = true
	} else if !(condition.Reason == reason && condition.Message == message &&
		condition.Commit == commit && condition.ErrorSourceRefs == errorSources &&
		condition.ErrorSummary == errorSummary) {
		updated = true
	} else {
		return updated, transitioned
	}
	condition.Reason = reason
	condition.Message = message
	condition.Commit = commit
	condition.ErrorSourceRefs = errorSources
	condition.ErrorSummary = errorSummary
        condition.LastUpdateTime = lastUpdate

	return updated, transitioned
```
IMO this demonstrates the intent more clearly and avoids the need for `DeepCopy`/`DeepEqual`. Could alternatively extract the boolean logic to a helper function. WDYT?

Reply:
Yeah, I was just feeling lazy and didn't want to write out all the comparisons... I'll paste yours in. But you could have used the github suggestion option to do it yourself and save me time ;).

Reply:
Cant == slices... bleh PITA to compare in Go. Can't just add an Equal method without extracting a list struct or making a custom function.

Reply:
Sigh. Can't == the timestamps either. They're pointers and we want to compare by value.

Reply:
Is this better?

```
if condition.Status != status {
		condition.Status = status
		condition.LastTransitionTime = lastUpdate
		transitioned = true
		updated = true
	} else if condition.Reason != reason ||
		condition.Message != message ||
		condition.Commit != commit ||
		!equality.Semantic.DeepEqual(condition.ErrorSourceRefs, errorSources) ||
		!equality.Semantic.DeepEqual(condition.ErrorSummary, errorSummary) {
		updated = true
	} else {
		return updated, transitioned
	}
	condition.Reason = reason
	condition.Message = message
	condition.Commit = commit
	condition.ErrorSourceRefs = errorSources
	condition.ErrorSummary = errorSummary
	condition.LastUpdateTime = lastUpdate

	return updated, transitioned
}
```

Reply:
Yeah that LGTM